extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts

```diff
@@ -0,0 +1,328 @@
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+
+import {
+    ChePluginRegistry,
+    ChePluginMetadata,
+    ChePluginService,
+    CheApiService
+} from '../../common/che-protocol';
+
+import { PluginServer } from '@theia/plugin-ext/lib/common/plugin-protocol';
+import { MessageService, Emitter, Event } from '@theia/core/lib/common';
+import { ConfirmDialog } from '@theia/core/lib/browser';
+
+import { ChePluginPreferences } from './che-plugin-preferences';
+import { ChePluginFrontentService } from './che-plugin-frontend-service';
+import { PreferenceService, PreferenceScope } from '@theia/core/lib/browser/preferences';
+
+@injectable()
+export class ChePluginManager {
+
+    /**
+     * Default plugin registry
+     */
+    private defaultRegistry: ChePluginRegistry;
+
+    /**
+     * Active plugin registry.
+     * Plugin widget should display the list of plugins from this registry.
+     */
+    private activeRegistry: ChePluginRegistry;
+
+    /**
+     * Registry list
+     */
+    private registryList: ChePluginRegistry[];
+
+    /**
+     * List of installed plugins received from workspace config.
+     */
+    private installedPlugins: string[];
+
+    /**
+     * List of plugins, currently available on active plugin registry.
+     */
+    private availablePlugins: ChePluginMetadata[] = [];
+
+    @inject(ChePluginService)
+    protected readonly chePluginService: ChePluginService;
+
+    @inject(PluginServer)
+    protected readonly pluginServer: PluginServer;
+
+    @inject(CheApiService)
+    protected readonly cheApiService: CheApiService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @inject(ChePluginPreferences)
+    protected readonly chePluginPreferences: ChePluginPreferences;
+
+    @inject(PreferenceService)
+    protected readonly preferenceService: PreferenceService;
+
+    @inject(ChePluginFrontentService)
+    protected readonly pluginFrontentService: ChePluginFrontentService;
+
+    protected readonly pluginRegistryChanged = new Emitter<ChePluginRegistry>();
+
+    protected readonly workspaceConfigurationChanged = new Emitter<boolean>();
+
+    get onPluginRegistryChanged(): Event<ChePluginRegistry> {
+        return this.pluginRegistryChanged.event;
+    }
+
+    get onWorkspaceConfigurationChanged(): Event<boolean> {
+        return this.workspaceConfigurationChanged.event;
+    }
+
+    /**
+     * Restores list of custom registries
+     */
+    private async restoreRegistryList(): Promise<void> {
+        // wait for preference service
+        await this.preferenceService.ready;
+
+        const prefs = this.chePluginPreferences['chePlugins.repositories'];
+        if (prefs) {
+            const instance = this;
```

Comment:
You don't need to save `this` as `instance` because an arrow function doesn't have its own `this`, and `this` of surrounding scope is used.

Reply:
Yes, it's already fixed. Thanks!